src/runtime_src/core/pcie/linux/shim.cpp

```diff
@@ -116,6 +116,149 @@ inline int io_getevents(aio_context_t ctx, long min_nr, long max_nr,
 
 namespace xocl {
 
+/*
+ * queue_cb: control block for a qdma queue
+ */
+class queue_cb {
+public:
+    friend class shim;
```

Comment:
This class is internal to the shim.cpp compilation unit (which is good), but whats the point of 'class' (vs struct) when shim is made a friend and shim all of this file.  In other words, maybe just make this a struct and forget about private encapsulation.